### DISCUSSION TITLE
Console Window Runs Outside SEB with Visual Studio

### DISCUSSION THREAD

**Discussion by KaggBrorson** [SafeExamBrowser/seb-win-refactoring#1027] [Q&A]
Hi,
I am trying to use Safe Exam Browser (SEB) to run Visual Studio for C# programming. Visual Studio works fine to start and use, but I am having an issue with the console window:

When I run a console application in Visual Studio, the console window opens and the program executes, but the console window appears outside SEB. As a result, it is not accessible while SEB is running.

When I close SEB, I can see the console windows in the background, showing that the programs executed correctly, but SEB seems to block or hide the windows.

I have tried changing the Kiosk Mode setting to both Create New Desktop and None (Debugging only), but the issue persists regardless of the mode.

**Here’s my setup:**
SEB Version: 3.8.0
The following applications are added under the Applications tab in SEB:
- msbuild.exe
- devenv.exe
- dotnet.exe
- cmd.exe
- powershell.exe

All applications are configured with "Allow running in background" and tested aoutostart.
Kiosk Mode is set to Disable Explorer Shell (I have also tested Create New Desktop and None (Debugging only)).

**Questions:**
1. Does anyone know how to make the console window appear and be accessible within SEB?
2. Is this a known issue with SEB and console applications?
3. Are there any specific settings or workarounds to resolve this?

Thank you in advance for any suggestions or help!

**Comment by dbuechel** [comment]
Try enabling the real time application log (see https://safeexambrowser.org/windows/win_usermanual_en.html#SecurityPane) and verify whether SEB is hiding the console window. If so, you'd simply need to add or fix the configuration in the list of permitted applications in your configuration.

**Comment by heremansgeert** [comment]
Did you ever managed to fix this issue? 

Think the problem is that's it is not possible to allow wilcards in the permitted application list. This way you could permit the students to run exe's in their repos folder.
